nntrainer/src/neuralnet.cpp

```diff
@@ -673,27 +697,25 @@ int NeuralNetwork::train(
  */
 int NeuralNetwork::train_run() {
   int status = ML_ERROR_NONE;
-  status = data_buffer->run(nntrainer::BUF_TRAIN);
-  if (status != ML_ERROR_NONE) {
-    data_buffer->clear(BUF_TRAIN);
-    return status;
-  }
 
-  status = data_buffer->run(nntrainer::BUF_VAL);
-  if (status != ML_ERROR_NONE) {
-    data_buffer->clear(BUF_VAL);
-    return status;
-  }
-
-  status = data_buffer->run(nntrainer::BUF_TEST);
-  if (status != ML_ERROR_NONE) {
-    data_buffer->clear(BUF_TEST);
-    return status;
+  if (data_buffer->getValidation()[2]) {
```

Comment:
Minor... IMO this could be changed to enum value
```suggestion
  if (data_buffer->getValidation()[BUF_TEST]) {
```